generated/typespec-ts/src/api/operations.ts

```diff
@@ -0,0 +1,1708 @@
+> {
+  return context
+    .path("/audio/translations")
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      contentType: (options.contentType as any) ?? "multipart/form-data",
+      body: {
+        file: uint8ArrayToString(audio["file"], "binary"),
+        model: audio["model"],
+        prompt: audio["prompt"],
+        response_format: audio["responseFormat"],
+        temperature: audio["temperature"],
+      },
+    });
+}
+
+export async function _createTranslationDeserialize(
+  result: CreateTranslation200Response | CreateTranslationDefaultResponse
+): Promise<CreateTranslationResponse> {
+  if (isUnexpected(result)) {
+    throw result.body;
+  }
+
+  return {
+    text: result.body["text"],
+  };
+}
+
+export async function createTranslation(
+  context: Client,
+  audio: CreateTranslationRequest,
+  options: CreateTranslationOptions = { requestOptions: {} }
+): Promise<CreateTranslationResponse> {
+  const result = await _createTranslationSend(context, audio, options);
+  return _createTranslationDeserialize(result);
+}
+
+export function _createChatCompletionSend(
+  context: Client,
+  body: CreateChatCompletionRequest,
+  options: CreateChatCompletionOptions = { requestOptions: {} }
+): StreamableMethod<
+  CreateChatCompletion200Response | CreateChatCompletionDefaultResponse
+> {
+  return context
+    .path("/chat/completions")
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      body: {
+        model: body["model"],
+        messages: (body["messages"] ?? []).map((p) => ({
+          role: p["role"],
+          content: p["content"],
+          name: p["name"],
+          function_call: !p.functionCall
+            ? undefined
+            : {
+                name: p.functionCall?.["name"],
+                arguments: p.functionCall?.["arguments"],
+              },
+        })),
+        functions: (body["functions"] ?? []).map((p) => ({
+          name: p["name"],
+          description: p["description"],
+          parameters: p["parameters"],
+        })),
+        function_call: body["functionCall"],
+        temperature: body["temperature"],
+        top_p: body["topP"],
+        n: body["n"],
+        max_tokens: body["maxTokens"],
+        stop: body["stop"],
+        presence_penalty: body["presencePenalty"],
+        frequency_penalty: body["frequencyPenalty"],
+        logit_bias: body["logitBias"],
+        user: body["user"],
+        stream: body["stream"],
+      },
+    });
+}
+
+export async function _createChatCompletionDeserialize(
+  result: CreateChatCompletion200Response | CreateChatCompletionDefaultResponse
+): Promise<CreateChatCompletionResponse> {
+  if (isUnexpected(result)) {
+    throw result.body;
+  }
+
+  return {
+    id: result.body["id"],
+    object: result.body["object"],
+    created: new Date(result.body["created"]),
+    model: result.body["model"],
+    choices: (result.body["choices"] ?? []).map((p) => ({
```

Comment:
[Compile Error]

src/api/operations.ts:278:5 - error TS2322: Type '{ index: any; message: { role: any; content: any; functionCall: { name: any; arguments: any; } | undefined; }; finishReason: any; }[]' is not assignable to type '[]'.
  Target allows only 0 element(s) but source may have more.

278     choices: (result.body["choices"] ?? []).map((p) => ({